### PR TITLE
fix: error handling, dark mode, consent bug fixes

### PR DESCRIPTION
**Pull request by siddhart1o1**

### CODE REVIEW

File: apps/consent/app/components/logo/index.tsx [hunk 1]
```diff
@@ -1,11 +1,62 @@
 import React from "react"
-import Image from "next/image"
 
 const Logo: React.FC = () => {
   return (
-    <div className="flex justify-center mb-4">
-      <Image src="/blink_logo.svg" alt="Galoy" width={120} height={120} />
-    </div>
+    <svg
```
Comment (nicolasburtey) [inline apps/consent/app/components/logo/index.tsx:5]: I think svg are best saved as .svg

File: apps/consent/app/error-handler.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+interface ErrorResponse {
+  error: boolean
+  message: string
+  responsePayload: null
+}
+
+const errorMessages: { [key: string]: string } = {
+  UserCodeAttemptIpRateLimiterExceededError:
+    "Your rate limit exceeded, please try after some time.",
```
Comment (nicolasburtey) [inline apps/consent/app/error-handler.ts:11]:
```suggestion
```

File: apps/consent/app/login/email-login-server-action.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import { cookies, headers } from "next/headers"
+
+import { hydraClient } from "../../services/hydra"
+import { LoginType, SubmitValue } from "../index.types"
+
+import { LoginEmailResponse } from "./email-login.types"
+
+import authApi from "@/services/galoy-auth"
+import { handleAxiosError } from "@/app/error-handler"
+//  this page is for login via email
```
Comment (nicolasburtey) [inline apps/consent/app/login/email-login-server-action.ts:12]:
```suggestion
```

Comment (nicolasburtey) [reply apps/consent/app/login/email-login-server-action.ts:12]: this comment doesn't really add value imo

File: apps/consent/app/error-handler.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  UserCodeAttemptIdentifierRateLimiterExceededError:
+    "Your rate limit exceeded, please try after some time.",
+}
+
+/* eslint @typescript-eslint/ban-ts-comment: "off" */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const handleAxiosError = (err: any): ErrorResponse => {
```
Comment (nicolasburtey) [inline apps/consent/app/error-handler.ts:18]:
```suggestion
export const handleAxiosError = (err: unknown): ErrorResponse => {
```

can you have `unknown` instead?

File: apps/consent/app/login/email-login-server-action.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    )
+    redirect(response.data.redirect_to)
+  }
+
+  if (!email || typeof email !== "string") {
+    console.error("Invalid Values for email")
+    throw new Error("Invalid Email Value")
+  }
+
+  let emailCodeRequest
```
Comment (nicolasburtey) [inline apps/consent/app/login/email-login-server-action.ts:61]: no implicit any

File: apps/consent/app/login/phone/server-actions.ts [hunk 6]
```diff
@@ -61,9 +70,41 @@ export const getCaptchaChallenge = async (
     }
   }
 
-  const res = await authApi.requestPhoneCaptcha(customHeaders)
+  let res
```
Comment (nicolasburtey) [inline apps/consent/app/login/phone/server-actions.ts:73]: no implicit any

@vindard could we enforce this in the apps?

File: apps/consent/env.ts [hunk 7]
```diff
@@ -8,10 +8,12 @@ export const env = createEnv({
   },
   shared: {
     GRAPHQL_ENDPOINT: z.string().default("http://localhost:4002/graphql"),
+    NODE_ENV: z.string().default("development"),
```
Comment (nicolasburtey) [inline apps/consent/env.ts:11]: cf comment on the other repo. 

it's not ideal to have several PR open for review with code overlapping. it adds overhead for the reviewer.

**Comment by siddhart1o1** [comment]
This PR will be split into smaller PRs.

**Pull request closed without merging**